ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java

```diff
@@ -192,6 +192,11 @@ default void notifyLogFailed(Throwable cause, LogEntryProto failedEntry) {}
      * completely done. Could trigger the cleanup of snapshots.
      */
     default void notifySnapshotInstalled(InstallSnapshotResult result, long snapshotIndex) {}
+
+    /**
+     * Notify the {@link StateMachine} that a raft server has step down.
+     */
+    default void notifyServerStepDown() {}
```

Comment:
Let's call it notifyServerShutdown and pass a RoleInfoProto parameter.

Reply:
done